### PR TITLE
issue: 1628881 Add ring modification function to vma_extra api

### PR DESCRIPTION
**Pull request by DanielLibenson**
Signed-off-by: Daniel Libenson <danielli@mellanox.com>

### CODE REVIEW

File: src/vma/vma_extra.h [hunk 1]
```diff
@@ -266,6 +266,33 @@ typedef enum {
...
+struct vma_cq_moderation_attr {
+	uint32_t cq_moderation_count;
+	uint32_t cq_moderation_period_usec;
+};
+
+struct vma_cq_arm_attr {
+};
+
+/**
+ * @param comp_mask - what fields should be read when processing this sturct
```
Comment (liranoz12) [inline src/vma/vma_extra.h]: typo - struct

File: src/vma/sock/sock-redirect.cpp [hunk 2]
```diff
@@ -677,6 +677,37 @@ int vma_add_ring_profile(vma_ring_type_attr *profile, vma_ring_profile_key *res)
 	return 0;
 }
 
+extern "C"
+int vma_modify_ring(struct vma_modify_ring_attr *mr_data)
+{
+	srdr_logdbg_entry("ring_fd=%d, mr_data=%p ", mr_data->ring_fd, mr_data);
```
Comment (liranoz12) [inline src/vma/sock/sock-redirect.cpp]: I am not sure about the dbg log level.

File: src/vma/sock/sock-redirect.cpp [hunk 3]
```diff
@@ -677,6 +677,37 @@ int vma_add_ring_profile(vma_ring_type_attr *profile, vma_ring_profile_key *res)
...
+	int ret = -1;
+	cq_channel_info* p_cq_ch_info = g_p_fd_collection->get_cq_channel_fd(mr_data->ring_fd);
+	if (likely(p_cq_ch_info)) {
+		ring_simple* p_ring = dynamic_cast<ring_simple*>(p_cq_ch_info->get_ring());
+		if (likely(p_ring)) {
+			if (VMA_MODIFY_RING_CQ_MODERATION & mr_data->comp_bit_mask) {
+				p_ring->modify_cq_moderation(mr_data->cq_moderation.cq_moderation_period_usec,
+						mr_data->cq_moderation.cq_moderation_count);
+				ret = 0;
```
Comment (liranoz12) [inline src/vma/sock/sock-redirect.cpp]: You are using the "ret" parameter here while below you return immediately...
please be consistent

File: src/vma/dev/ring.h [hunk 4]
```diff
@@ -54,7 +54,8 @@ class ring_slave;
 
 typedef enum {
 	CQT_RX,
-	CQT_TX
+	CQT_TX,
+	CQT_RX_TX
```
Comment (liranoz12) [inline src/vma/dev/ring.h]: I dont like this new parameter hack.
Can we use ring_slave:m_type in order to decide which cq_type we should use ?

File: src/vma/sock/sock-redirect.cpp [hunk 5]
```diff
@@ -677,6 +677,37 @@ int vma_add_ring_profile(vma_ring_type_attr *profile, vma_ring_profile_key *res)
...
+			} else if (VMA_MODIFY_RING_CQ_ARM & mr_data->comp_bit_mask) {
+				return p_ring->request_notification(CQT_RX_TX, 0);
+			} else {
+				vlog_printf(VLOG_ERROR, "comp_mask [0x%x] is not supported\n",
```
Comment (liranoz12) [inline src/vma/sock/sock-redirect.cpp]: Please use srdr_logerr macro - no need to add "\n" with this macro

File: src/vma/vma_extra.h (hunk 1 above)
Comment (DanielLibenson) [reply src/vma/vma_extra.h]: :+1:

File: src/vma/sock/sock-redirect.cpp (hunk 3 above)
Comment (DanielLibenson) [reply src/vma/sock/sock-redirect.cpp]: :+1:

File: src/vma/dev/ring.h (hunk 4 above)
Comment (DanielLibenson) [reply src/vma/dev/ring.h]: :+1:

File: src/vma/sock/sock-redirect.cpp (hunk 5 above)
Comment (DanielLibenson) [reply src/vma/sock/sock-redirect.cpp]: :-1:

**Comment by DanielLibenson** [comment]
bot:retest

**Comment by DanielLibenson** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Pull request merged**